### PR TITLE
✨ Add Delete Process Model Functionality

### PR DESCRIPTION
**Pull request by SteffenKn**
**Changes:**

1. Add delete process model functionality

**Issues:**

Related to https://github.com/process-engine/process_engine_runtime/issues/64

PR: #22

## How can others test the changes?

- **Have a look at https://github.com/process-engine/process_engine_runtime/pull/141**

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Fixing Security Issues   | `:lock:`             | 🔒     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️      |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️      |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: src/accessors/internal_accessor.ts [hunk 1]
```diff
@@ -121,6 +121,10 @@ export class InternalAccessor implements IManagementApiAccessor {
     return this._managementApiService.updateProcessDefinitionsByName(identity, name, payload);
   }
 
+  public async deleteProcessDefinitionsByProcessModelId(identity: IIdentity, name: string): Promise<void> {
```
Comment (ElRaptorus) [inline src/accessors/internal_accessor.ts]: Please use `processModelId` instead of `name`

File: src/management_api_client_service.ts [hunk 2]
```diff
@@ -103,6 +103,13 @@ export class ManagementApiClientService implements IManagementApi {
     return this.managementApiAccessor.updateProcessDefinitionsByName(identity, name, payload);
   }
 
+  public async deleteProcessDefinitionsByProcessModelId(identity: IIdentity,
+                                                        processModelId: string,
+                                                    ): Promise<void> {
```
Comment (ElRaptorus) [inline src/management_api_client_service.ts]: This alignment looks weird.

If you want to use line breaks, please keep the round brackets aligned.

But unless tslint complains, I'd just place the signature all on the same line.

File: src/management_api_client_service.ts [hunk 3]
```diff
@@ -97,12 +97,18 @@ export class ManagementApiClientService implements IManagementApi {
...
-                                              payload: ProcessModelExecution.UpdateProcessDefinitionsRequestPayload,
-                                          ): Promise<void> {
+                                              payload: ProcessModelExecution.UpdateProcessDefinitionsRequestPayload): Promise<void> {
 
     return this.managementApiAccessor.updateProcessDefinitionsByName(identity, name, payload);
   }
 
+  public async deleteProcessDefinitionsByProcessModelId(identity: IIdentity,
+                                                        processModelId: string,
+                                                    ): Promise<void> {
```
Comment (ElRaptorus) [inline src/management_api_client_service.ts]: Please fix the alignment. 
Just write 
`public async deleteProcessDefinitionsByProcessModelId(identity: IIdentity, processModelId: string): Promise<void> {`

TSLint shouldn't complain.

Comment (SteffenKn) [reply src/management_api_client_service.ts]: I fixed the wrong indentation in `💄 Fix Indentation`...

**Pull request merged**